include/swift/AST/StorageImpl.h

```diff
@@ -355,6 +355,13 @@ class StorageImplInfo {
             getWriteImpl() == WriteImplKind::Immutable);
   }
 
+  bool isSimplyStoredLazy() const {
+    return getReadImpl() == ReadImplKind::Stored &&
+           (getWriteImpl() == WriteImplKind::Stored ||
+            getWriteImpl() == WriteImplKind::Immutable ||
```

Comment:
I don't think you want to allow 'Immutable' here either; let properties cannot be lazy (or have observers).